File: gdx2d-demoDesktop/src/ch/hevs/gdx2d/demos/physics/pinball/ScoreAnimation.java
package ch.hevs.gdx2d.demos.physics.pinball;

import ch.hevs.gdx2d.lib.GdxGraphics;
import com.badlogic.gdx.Gdx;
import com.badlogic.gdx.graphics.Color;
import com.badlogic.gdx.utils.Align;

public class ScoreAnimation implements TemporaryDrawable {

	private float duration;
	private String msg;
	private float x;
	private float y;
	private float start_duration;

	public ScoreAnimation(String msg, float x, float y) {
		this.start_duration = 1.0f;
		this.duration = 1.0f;
		this.msg = msg;
		this.x = x;
		this.y = y;
	}

	@Override
	public void draw(GdxGraphics g) {
		duration -= Gdx.graphics.getDeltaTime();
		g.setColor(new Color(Color.BLACK).mul(duration/start_duration));
		g.drawString(x, y, msg, Align.center);
	}

	@Override
	public boolean isDone() {
		return duration < 0;
	}
}

File: gdx2d-demoDesktop/src/main/scala/ch/hevs/gdx2d/demos/physics/pinball/ScoreAnimation.scala
package ch.hevs.gdx2d.demos.physics.pinball

import ch.hevs.gdx2d.lib.GdxGraphics
import com.badlogic.gdx.Gdx
import com.badlogic.gdx.graphics.Color
import com.badlogic.gdx.utils.Align

class ScoreAnimation(private val msg: String, private val x: Float, private val y: Float) extends TemporaryDrawable {

  private var duration: Float = 1.0f
  private val startDuration: Float = 1.0f

  override def isDone: Boolean = duration < 0

  override def draw(g: GdxGraphics): Unit = {
    duration -= Gdx.graphics.getDeltaTime
    g.setColor(new Color(Color.BLACK).mul(duration / startDuration))
    g.drawString(x, y, msg, Align.center)
  }
}
